### ISSUE TITLE
[plugins] gemspec files includes all vendor/ files 

### ISSUE THREAD

**Issue by colinsurprenant** [elastic/logstash#4242]
most plugins have this gemspec **files** specification
```
  s.files = Dir['lib/**/*','spec/**/*','vendor/**/*','*.gemspec','*.md','CONTRIBUTORS','Gemfile','LICENSE','NOTICE.TXT']
```

the `vendor/**/*` pattern will include all bundled gems if/when building the gem locally when using the "standard" `vendor/` bundle path config. 

for some plugins, some files in `vendor/` need to be included, like the geoip database in the geoip filter or jar files for other plugins. 

we could either use other directories for non-bundle gems vendored files like `vendor/data` and `vendor/jar` ? or we could exclude the gems specific directories in `vendor/` :
```
drwxr-xr-x+ 18 colin staff  612 Nov 12  2014 bin
drwxr-xr-x+ 63 colin staff 2.1K Nov 12  2014 build_info
drwxr-xr-x+ 63 colin staff 2.1K Nov 12  2014 cache
drwxr-xr-x+  2 colin staff   68 Nov 12  2014 doc
drwxr-xr-x+ 63 colin staff 2.1K Nov 12  2014 gems
drwxr-xr-x+ 63 colin staff 2.1K Nov 12  2014 specifications
``` 

@ph is this files specification normally standard across all plugins and updated with the plugins mass update scripts?

thoughts?

**Comment by ph** [comment]
This is similar to the other issue I have created earlier https://github.com/elastic/logstash/issues/4241, no problem I will close it.

@colinsurprenant Concerning the `s.files` I can say its "standard" accross all the plugins that we have in `logstash-plugins` since the last mass update from @jsvd, things may change in the future since we are opening the doors to the contributors. But for now its the same.

I prefer to be more explicit about what we bundle in the gem instead of removing stuff. So `vendor/jars` or `vendor/data` seem like a good choice to me.

**Comment by ph** [comment]
So following that we could update it with:
```
 s.files = Dir['lib/**/*','spec/**/*','vendor/data/**/*','*.gemspec','*.md','CONTRIBUTORS','Gemfile','LICENSE','NOTICE.TXT']
```
I thought the jars were in the `lib` to make sure they were added to the `$LOAD_PATH`

**Comment by colinsurprenant** [comment]
@ph yeah, you are right about the jars, at least for the ones I looked at.
+1 on including instead of excluding.
we will have to go through all plugins to verify/update any vendored files paths.

**Comment by colinsurprenant** [comment]
oh, I didn't see #4241 :P

**Comment by ph** [comment]
they are easy to catch, search for `vendor.json`

```
logstash-codec-collectd/Rakefile
logstash-filter-geoip/Rakefile
logstash-filter-useragent/Rakefile
```
